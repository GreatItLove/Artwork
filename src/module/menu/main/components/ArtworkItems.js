import React from 'react';
import { ListItem } from 'material-ui/List';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

import { ARTWORK_PATH } from '../../../../config/routesPatch';

const propTypes = {
  style: PropTypes.object.isRequired
};

/**
 * @function ArtworkItems
 * @param {Object} style
 * @returns {XML}
 */
function ArtworkItems({ style }) {
  return (
    <ListItem
      key={ARTWORK_PATH}
      primaryText="Artwork"
      innerDivStyle={style}
      leftIcon={<i className="material-icons">color_lens</i>}
      containerElement={<NavLink to={ARTWORK_PATH} exact />}
    />
  );
}

ArtworkItems.propTypes = propTypes;

export default ArtworkItems;
