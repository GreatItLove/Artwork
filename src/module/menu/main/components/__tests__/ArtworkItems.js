import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import ArtworkItems from '../ArtworkItems';

describe('ArtworkItems', () => {
  it('Snapshot of ArtworkItems', () => {
    const wrapper = shallow(<ArtworkItems style={{}} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
