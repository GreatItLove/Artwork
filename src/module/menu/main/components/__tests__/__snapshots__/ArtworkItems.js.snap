// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtworkItems Snapshot of ArtworkItems 1`] = `
<ListItem
  autoGenerateNestedIndicator={true}
  containerElement={
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      exact={true}
      to="/artwork"
    />
  }
  disableKeyboardFocus={false}
  disabled={false}
  initiallyOpen={false}
  innerDivStyle={Object {}}
  insetChildren={false}
  leftIcon={
    <i
      className="material-icons"
    >
      color_lens
    </i>
  }
  nestedItems={Array []}
  nestedLevel={0}
  onKeyboardFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onNestedListToggle={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  open={null}
  primaryText="Artwork"
  primaryTogglesNestedList={false}
  secondaryTextLines={1}
/>
`;
